### ISSUE TITLE
Structured logs: remaining bits

### ISSUE THREAD

**Issue by orpiske** [apache/camel-k#2286]
This is a follow up ticket from #2279 and it is related to the remaining bits on our code base and/or dependencies that don't provide support for structured logs. 

Some parts of the code that currently don't have structured logs: 

leader election code from k8s:

```
I0512 12:46:29.777593       1 leaderelection.go:243] attempting to acquire leader lease  test-aa6ed267-c505-4180-9618-eee3f53a04c6/camel-k-lock...
I0512 12:46:29.781821       1 leaderelection.go:253] successfully acquired lease test-aa6ed267-c505-4180-9618-eee3f53a04c6/camel-k-lock
```

k8s warnings:
```
W0512 12:54:23.468230       1 warnings.go:67] extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
```

This list may include other items in the future.

**Comment by orpiske** [comment]
Quoting @astefanutti here for convenience:

> ...
> Indeed, structured logging is still in alpha in Kubernetes 1.19:
> https://kubernetes.io/blog/2020/09/04/kubernetes-1-19-introducing-structured-logs/. It's also not really clear how to activate  structured logging in client-go (which hosts the leader election code that controller-runtime calls), maybe klog.SetLogger() can be used?
> ...

**Comment by astefanutti** [comment]
Fixed with #2286.
